source/Calamari.Tests/Fixtures/PackageRetention/LeastFrequentlyUsedWithAgingCacheAlgorithmFixture.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Calamari.Common.Plumbing.Deployment.PackageRetention;
+using Calamari.Deployment.PackageRetention.Caching;
+using Calamari.Deployment.PackageRetention.Model;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Calamari.Tests.Fixtures.PackageRetention
+{
+    [TestFixture]
+    public class LeastFrequentlyUsedWithAgingCacheAlgorithmFixture
+    {
+        [Test]
+        public void WhenSpaceNeededIsGreaterThanSpaceUsed_ThenThrowException()
+        {
+            var lfu = new LeastFrequentlyUsedWithAgingCacheAlgorithm();
+            Assert.Throws<InsufficientCacheSpaceException>(() => lfu.GetPackagesToRemove(new List<JournalEntry>(), 10_000));
+        }
+
+        [Test]
+        public void WhenPackageIsLocked_ThenDoNotConsiderItForRemoval()
+        {
+            var lfu = new LeastFrequentlyUsedWithAgingCacheAlgorithm();
+            var entry = CreateEntry("package-1", "1.0", 20, ("task-1", 2));
+            entry.AddLock(new ServerTaskId("task-0"), new CacheAge(1));
+
+            Assert.Throws<InsufficientCacheSpaceException>(() => lfu.GetPackagesToRemove(new List<JournalEntry>(), 10));
```

Comment:
I don't think I understand this test. Does `ThenDoNotConsiderItForRemoval` mean an exception is thrown because there are no packages eligible for removal? If it was eligible, there would not be an exception?

Reply:
It would be clearer if there were two packages considered (one locked), with the locked package as the 'better' option, and we checked that the non-locked package was returned.  I'll update it. 

As I've coded it here, if the locked package wasn't locked, it would be returned because it would free up enough space.  Given that it *is* locked, we get an exception, because we don't have anything we can remove to free the required space.

Reply:
Oooooh, I see - I didn't actually add the entry to the list 🤦‍♂️
I'll do that, and also make the change I've mentioned above.